src/providers/buyProvider/BuyProvider.tsx

```diff
@@ -0,0 +1,72 @@
+import {
+  createContext,
+  FC,
+  PropsWithChildren,
+  useContext,
+  useEffect,
+  useReducer,
+} from "react";
+import { Action, InitialState, assetsBuyContext } from "./types";
+import { ASSETS_TRANSAK } from "../../constants/assets-transak";
+import { transformAddress } from "@src/utils/account-utils";
+
+const initialState: InitialState = {
+  chains: [],
+};
+
+export const reducer = (state: InitialState, action: Action): InitialState => {
+  switch (action.type) {
+    case "init-chains": {
+      const { chains } = action.payload;
+      return {
+        ...state,
+        chains,
+      };
+    }
+    default:
+      return state;
+  }
+};
+const BuyContext = createContext({} as assetsBuyContext);
+
+export const BuyProvider: FC<PropsWithChildren> = ({ children }) => {
```

Comment:
This provider is not necessary for the moment, It can be converted to a hook. For example:

```
const { createOrder, chains } = useBuy()
```